packages/react/src/components/FileUploader/FileUploaderItem.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { settings } from 'carbon-components';
+import classNames from 'classnames';
+import { Filename } from './FileUploader';
+import { keys, matches } from '../../internal/keyboard';
+
+const { prefix } = settings;
+
+export default function FileUploaderItem({
+  uuid,
+  name,
+  status,
+  iconDescription,
+  onDelete,
+  invalid,
+  errorSubject,
+  errorBody,
+  ...other
+}) {
+  const classes = classNames(`${prefix}--file__selected-file`, {
+    [`${prefix}--file__selected-file--invalid`]: invalid,
+  });
+  return (
+    <span className={classes} {...other}>
+      <p className={`${prefix}--file-filename`}>{name}</p>
+      <span className={`${prefix}--file__state-container`}>
+        <Filename
+          iconDescription={iconDescription}
+          status={status}
+          onKeyDown={evt => {
+            if (matches(evt, [keys.Enter, keys.Space])) {
+              if (status === 'edit') {
+                onDelete(evt, { uuid });
+              }
+            }
+          }}
+          onClick={evt => {
+            if (status === 'edit') {
+              onDelete(evt, { uuid });
+            }
+          }}
+        />
+      </span>
+      {invalid && (
```

Comment:
Do we anticipate empty `errorSubject`/`errorBody` while `invalid`? Just wanted to double-check.

Reply:
I think we should consider making `errorSubject` and `errorBody` required props but I've added a check for those props before rendering now